### PR TITLE
`loans` - add collateral conversion and confirm page

### PR DESCRIPTION
**Pull request by thedoublejay**
#### What kind of PR is this?:

/kind feature

#### What this PR does / why we need it:
1. Add Collateral confirm page
2. Conversion flow for Add Collateral
#### Which issue(s) does this PR fixes?:

Fixes #

#### Developer Checklist:

- [ ] Read your code changes at least once
- [ ] Tested on iOS/Android device (e.g, No crashes, library supported etc.)
- [ ] No console errors on web
- [ ] Tested on Light mode and Dark mode*
- [ ] Your UI implementation visually matched the rendered design*
- [ ] Unit tests*
- [ ] Added e2e tests*
- [ ] Added translations*

### CODE REVIEW

File: mobile-app/app/screens/AppNavigator/screens/Loans/screens/ConfirmEditCollateralScreen.tsx [hunk 1]
```diff
@@ -140,95 +182,63 @@ function SummaryTransactionDetails (props: {tokensToAdd: number, totalCollateral
...
       <NumberRow
-        lhs={translate('screens/ConfirmEditCollateralScreen', 'Collateral amount (USD)')}
+        lhs={translate('screens/ConfirmEditCollateralScreen', 'Collateral amount')}
         rhs={{
-          value: new BigNumber(props.collateralAmount).toFixed(8),
+          value: props.amount.toFixed(8),
           testID: 'collateral_amount',
           suffixType: 'text',
-          suffix: ` ${props.collateralId}`
+          suffix: ` ${props.token.displaySymbol}`
```
Comment (kyleleow) [inline mobile-app/app/screens/AppNavigator/screens/Loans/screens/ConfirmEditCollateralScreen.tsx:221]:
```suggestion
          suffix: `${props.token.displaySymbol}`
```

File: mobile-app/app/screens/AppNavigator/screens/Loans/screens/ConfirmEditCollateralScreen.tsx [hunk 2]
```diff
@@ -6,49 +6,87 @@ import BigNumber from 'bignumber.js'
...
+      }, () => {
+        dispatch(fetchVaults({
+          address,
+          client
+        }))
+      })
+    }
   }
 
   function getSubmitLabel (): string {
```
Comment (kyleleow) [inline mobile-app/app/screens/AppNavigator/screens/Loans/screens/ConfirmEditCollateralScreen.tsx:92]: need to handle the condition to return `CONVERTING` label

**Review by kyleleow — APPROVED** [review]

File: mobile-app/app/screens/AppNavigator/screens/Loans/screens/ConfirmEditCollateralScreen.tsx (hunk 1 above)
Comment (thedoublejay) [reply mobile-app/app/screens/AppNavigator/screens/Loans/screens/ConfirmEditCollateralScreen.tsx:221]: will update

File: mobile-app/app/screens/AppNavigator/screens/Loans/screens/EditCollateralScreen.tsx [hunk 3]
```diff
@@ -104,16 +111,49 @@ export function EditCollateralScreen ({
...
-      token: item.token
-    }, dispatch, logger, () => {
-      dispatch(fetchVaults({
-        address,
-        client
-      }))
-    })
+    const isConversionRequired = item.token.id === '0' ? new BigNumber(item.amount).gt(DFIToken.amount) : false
+    const collateralItem = collateralTokens.find((col) => col.token.id === item.token.id)
+    if (activeVault !== undefined && collateralItem !== undefined) {
```
[the remark this answers is not in the document]
Comment (thedoublejay) [reply mobile-app/app/screens/AppNavigator/screens/Loans/screens/EditCollateralScreen.tsx:116]: will update

File: mobile-app/app/screens/AppNavigator/screens/Loans/screens/ConfirmEditCollateralScreen.tsx (hunk 1 above)
Comment (thedoublejay) [reply mobile-app/app/screens/AppNavigator/screens/Loans/screens/ConfirmEditCollateralScreen.tsx:221]: will update

File: mobile-app/app/screens/AppNavigator/screens/Loans/screens/ConfirmEditCollateralScreen.tsx (hunk 2 above)
Comment (thedoublejay) [reply mobile-app/app/screens/AppNavigator/screens/Loans/screens/ConfirmEditCollateralScreen.tsx:92]: will update

File: mobile-app/app/screens/AppNavigator/screens/Loans/screens/EditCollateralScreen.tsx [hunk 4]
```diff
@@ -57,20 +55,22 @@ export interface CollateralItem extends CollateralToken {
...
+  const DFIUtxo = useSelector((state: RootState) => DFIUtxoSelector(state.wallet))
+  const DFIToken = useSelector((state: RootState) => DFITokenSelector(state.wallet))
   const tokens = useTokensAPI()
   const getTokenAmount = (tokenId: string): BigNumber => {
-    return new BigNumber(tokens.find((t) => t.id === tokenId)?.amount ?? 0)
+    const id = tokenId === '0' ? '0_unified' : tokenId
+    return new BigNumber(tokens.find((t) => t.id === id)?.amount ?? 0)
   }
 
   const {
```
[the remark this answers is not in the document]
Comment (thedoublejay) [reply mobile-app/app/screens/AppNavigator/screens/Loans/screens/EditCollateralScreen.tsx:76]: You can, it's a `createSelector` so you can put as many operations as you want

**Pull request merged**